src/sys/node/node-worker-thread.ts

```diff
@@ -22,30 +29,30 @@ export const initNodeWorkerThread = (prcs: NodeJS.Process, msgHandler: d.WorkerM
         errMsgBackToMain.stencilRtnError += ':' + err.message;
       }
     }
-    prcs.send(errMsgBackToMain, sendHandle);
+    process.send(errMsgBackToMain, sendHandle);
   };
 
-  prcs.on('message', async (msgToWorker: d.MsgToWorker) => {
+  process.on('message', async <T extends d.WorkerContextMethod>(msgToWorker: d.MsgToWorker<T>) => {
     // message from the main thread
     if (msgToWorker && typeof msgToWorker.stencilId === 'number') {
       try {
         // run the handler to get the data
-        const msgFromWorker: d.MsgFromWorker = {
+        const msgFromWorker = {
           stencilId: msgToWorker.stencilId,
           stencilRtnValue: await msgHandler(msgToWorker),
-          stencilRtnError: null,
-        };
+          stencilRtnError: null as string | null,
+        } satisfies d.MsgFromWorker<T>;
```

Comment:
I may be missing something here, but couldn't we avoid the type assertion on `stencilRtnError` by explicitly typing `msgFromWorker` like we currently do on `main`?

Reply:
yeah maybe it's just easier to leave it that way, I'll put it back